### DISCUSSION TITLE
How to deal with seq_len dimmension with GCNConv?

### DISCUSSION THREAD

**Discussion by Percibus** [pyg-team/pytorch_geometric#9441] [Q&A] [answered]
I've just starting using PyG, and for my problem I have batched data with a time dimmension. My tensor has shape `[batch, seq_len, nodes, features]` . I want to apply the `GCNConv` layer separatly to each seq_len step.  

I read on this [issue](https://github.com/pyg-team/pytorch_geometric/discussions/4107)

> you need to permute your matrix to shape [batch_size, time_seq, num_nodes, num_feat]

But I'm not sure if I'm getting it correctly. Can I pass the tensor with this shape directly to the GCNConv layer and it will be applied to each seq_len step independenly, or should I reshape to `[batch * nodes, seq_len * features]` before feeding it into the layer? 

Thanks in advance for the help!

**Comment by rusty1s** [comment, marked as answer]
Yes, you can pass this tensor directly, and both `batch` and `seq_len` dimensions are treated separately.

**Comment by Percibus** [comment]
Hi, thank you for your time! I assume is the same as reshaping to `[batch * seq_len  , nodes , features]` as long as I use the default `node_dim = -2` ? 
And, when using operators that doesn't support batching directly, like `GATv2Conv`, I'm using:
```
x = x.view(batch_size * seq_len, nodes, features) # 48 nodes with 3 features each
data_list = [Data(x=x[i], edge_index=edge_index) for i in range(x.size(0))]
batch = Batch.from_data_list(data_list)  # and then pass batch.x and batch.edge_index to GAT      
``` 
Is this the correct approach to ensure each frame is processed as a distinct graph?
Thanks again for your answer.

**Reply by rusty1s** [reply]
Sorry for late reply. Yes, this is fully correct.

**Reply by Percibus** [reply]
Thanks!
